# Default configuration
--- ./site-sample.config	2003-09-03 14:09:14.000000000 +0200
+++ ./site-sample	2005-08-07 17:12:40.000000000 +0200
@@ -86,4 +86,4 @@
 #
 # Define where Integrit is installed (if you use it)
 #
-#INTEGRIT=/usr/local/bin/aide
+#INTEGRIT=/usr/local/bin/integrit
--- ./tigerrc-TAMU.config	2002-06-14 10:51:31.000000000 +0200
+++ ./tigerrc-TAMU	2005-08-07 17:12:40.000000000 +0200
@@ -15,6 +15,7 @@
 # you don't want performed.
 #
 Tiger_Check_PASSWD=Y		# Fast
+Tiger_Check_PASSWD_FORMAT=N     # Fast - not needed if on systems with pwck
 Tiger_Check_GROUP=Y		# Fast
 Tiger_Check_ACCOUNTS=Y		# Time varies on # of users
 Tiger_Check_RHOSTS=Y		# Time varies on # of users
@@ -141,6 +142,11 @@
 #
 Tiger_Crack_Local=N
 #
+# Who sends output from 'tigercron'?
+# Default is "root@$HOSTNAME" (gets expanded by tigercron)
+#
+# Tiger_Mail_FROM="root@`uname -n`"
+#
 # Who gets output from 'tigercron'?
 #
 Tiger_Mail_RCPT=root
--- ./cronrc.config	2003-09-19 16:24:41.000000000 +0200
+++ ./cronrc	2005-08-07 17:12:40.000000000 +0200
@@ -28,7 +28,16 @@
 # ----------------------------------------------------------------------
 # Check for known intrusion signs every 8 hours
 #
-0,8,16 * *	check_known check_rootkit check_finddeleted check_logfiles check_runprocs check_rootdir check_root
+0,8,16 * *	check_known check_rootkit check_logfiles check_runprocs check_rootdir check_root
+# [experimental]
+# Check_finddeleted is very verbose and needs to be adjusted so we run
+# it less often, admins that want to run it more often are suggested
+# to look up how does the ignore mechanism works. In busy servers many
+# false positives might appear because of normal user access.
+# Some applications might generate false positives too due to the 
+# way they behave.  Also, this script requires LSOF, so only enable
+# it if you have it installed.
+#1 * * check_finddeleted 
 # ----------------------------------------------------------------------
 # Make system-specific checks every day at 1 am
 # Notice: System specific checks can be enabled in the 
@@ -68,7 +77,7 @@
 # ----------------------------------------------------------------------
 # Check for system configuration once a month
 #
-1 2 *         check_services check_signatures check_umask check_ftpusers check_embedded check_exrc
+1 2 *         check_services check_umask check_ftpusers check_embedded check_exrc
 #
 # ----------------------------------------------------------------------
 # Run a password cracker against the local passwords once a month
--- ./tigerrc-dist.config	2003-05-05 15:14:34.000000000 +0200
+++ ./tigerrc-dist	2005-08-07 17:12:40.000000000 +0200
@@ -9,6 +9,7 @@
 #
 TigerNoBuild=Y			# C files are corrupted (ouch.)
 Tiger_Check_PASSWD=Y		# Fast
+Tiger_Check_PASSWD_FORMAT=N     # Fast - not needed if on systems with pwck
 Tiger_Check_PASSWD_SHADOW=Y	# Time varies on # of users
 Tiger_Check_PASSWD_NIS=N	# Time varies on # of users
 Tiger_Check_GROUP=Y		# Fast
@@ -209,6 +210,11 @@
 #
 Tiger_Crack_Local=Y
 #
+# Who sends output from 'tigercron'?
+# Default is "root@$HOSTNAME" (gets expanded by tigercron)
+#
+# Tiger_Mail_FROM="root@`uname -n`"
+#
 # Who gets output from 'tigercron'?
 #
 Tiger_Mail_RCPT=root
--- ./check.tbl.config	2002-06-14 10:51:31.000000000 +0200
+++ ./check.tbl	2005-08-07 17:12:40.000000000 +0200
@@ -22,4 +22,4 @@
 # Add other checks that should be attempted for all systems
 # here.  One script name per line.
 #
-check_sendmail
+#check_sendmail
--- ./tigerrc.config	2003-10-07 00:18:12.000000000 +0200
+++ ./tigerrc	2005-08-07 17:13:26.000000000 +0200
@@ -7,8 +7,16 @@
 # Select checks to perform.  Specify 'N' (uppercase) for checks
 # you don't want performed.
 #
+# Notice this does not affect which checks will be performed through
+# the cron job (Tiger's cronrc file governs that, usually under
+# /usr/local/etc/tiger or /etc/tiger), it only affects which
+# checks will be performed when running the full security checks
+# (i.e. 'tiger')
+#
+#
 TigerNoBuild=Y			# C files are corrupted (ouch.)
 Tiger_Check_PASSWD=Y		# Fast
+Tiger_Check_PASSWD_FORMAT=N     # Fast - not needed if on systems with pwck
 Tiger_Check_PASSWD_SHADOW=Y	# Time varies on # of users
 Tiger_Check_PASSWD_NIS=N	# Time varies on # of users
 Tiger_Check_GROUP=Y		# Fast
@@ -19,11 +27,11 @@
 Tiger_Check_CRON=Y		# Fast
 Tiger_Check_ANONFTP=Y		# Fast
 Tiger_Check_EXPORTS=Y		# Fast
-Tiger_Check_INETD=Y		# Fast
+Tiger_Check_INETD=Y		# Fast for inetd, Varies on xinetd
 Tiger_Check_SERVICES=Y		# Could be faster, not bad though
 Tiger_Check_KNOWN=Y		# Fast
 Tiger_Check_PERMS=Y		# Could be faster, not bad though
-Tiger_Check_SIGNATURES=Y	# Several minutes
+Tiger_Check_SIGNATURES=N	# Several minutes
 Tiger_Check_FILESYSTEM=Y	# Time varies on disk space... can be hours
 Tiger_Check_ROOTDIR=Y           # Fast, only 2 checks
 Tiger_Check_ROOT_ACCESS=Y       # Fast
@@ -37,7 +45,7 @@
 Tiger_Check_LISTENING=Y		# Fast
 Tiger_Check_SYSTEM=Y		# Depends on the specific system checks
 Tiger_Check_RUNPROC=N		# Fast, needs to be customized per system
-Tiger_Check_DELETED=Y		# Depends on the number of processes on the
+Tiger_Check_DELETED=N		# Depends on the number of processes on the
 				# system
 Tiger_Check_APACHE=N		# Fast
 Tiger_Check_SSH=Y		# Fast
@@ -46,6 +54,9 @@
 Tiger_Check_EXRC=N		# Depends on the size of the filesystem
 Tiger_Check_ROOTKIT=Y		# Slow if chkrootkit is available
 Tiger_Check_FTPUSERS=Y		# Fast
+Tiger_Check_OMNIBACK=N  	# Fast
+Tiger_Check_NTP=Y 		# Fast
+
 # OS specific checks
 # You can comment them if they are not appropiate to your system but
 # they will not run if you are running a different OS
@@ -181,7 +192,7 @@
 # What accounts are considered administrative (beyond root)
 # (likely to not be used by humans, and therefore have impossible passwords)
 # List of usernames separated by '|'... no whitespaces
-Tiger_Admin_Accounts='adm|bin|daemon|games|lp|mail|news|operator|sync|sys|uucp|man|proxy|majordom|postgres|www-data|operator|irc|gnats'
+Tiger_Admin_Accounts='bin|daemon|adm|lp|sync|shutdown|halt|mail|news|uucp|operator|games|gopher|ftp|nobody|dbus|vcsa|nscd|rpm|haldaemon|netdump|sshd|rpc|rpcuser|nfsnobody|mailnull|smmsp|pcap|apache|xfs|ntp|gdm|postfix|cacti|mach|mysql|named|clamav|zope|openvpn|dovecot'
 #
 # If an embedded pathname refers to an executable file, this executable
 # will in turn be checked.  This will continue "recursively" until
@@ -219,7 +230,7 @@
 # Who do you allow to own system files.
 # List of usernames separated by '|'... no whitespace
 #
-Tiger_Embedded_OK_Owners='root|bin|uucp|sys|daemon'
+Tiger_Embedded_OK_Owners='root|bin|uucp|sys|daemon|rpm'
 #Tiger_Embedded_OK_Owners=root
 #
 # What groups can have write access to system files?
@@ -238,7 +249,7 @@
 # Who can own executables in 'root's PATH?
 # List of usernames separated by '|'... no whitespace
 #
-Tiger_ROOT_PATH_OK_Owners='root|uucp|bin|news|sys|daemon'
+Tiger_ROOT_PATH_OK_Owners='root|uucp|bin|news|sys|daemon|rpm'
 # If you are paranoid:
 #Tiger_ROOT_PATH_OK_Owners='root'
 # If you are running HP-UX
@@ -253,13 +264,13 @@
 # Who can own things in other users PATH?
 # List of usernames separated by '|'... no whitespace
 #
-Tiger_PATH_OK_Owners='root|bin|daemon|uucp|sys|adm'
+Tiger_PATH_OK_Owners=$Tiger_ROOT_PATH_OK_Owners
 #
 # What groups can have write access to executables in non-root user PATH?
 # List of group names separated by '|'... no whitespace.
 # No value means no groups should have write access.
 #
-Tiger_PATH_OK_Group_Write=
+eval Tiger_PATH_OK_Group_Write='$Tiger_ROOT_PATH_OK_Group_Write'
 #
 # Should 'tiger' wait for Crack to finish?  If set to 'Y' it will wait
 # until it finishes.  If set to 'N', it will collect the output if
@@ -274,6 +285,11 @@
 #
 Tiger_Crack_Local=Y
 #
+# Who sends output from 'tigercron'?
+# Default is "root@$HOSTNAME" (gets expanded by tigercron)
+#
+# Tiger_Mail_FROM="root@`uname -n`"
+#
 # Who gets output from 'tigercron'?
 #
 Tiger_Mail_RCPT=root
@@ -285,12 +301,12 @@
 #
 # File system scan - things to look for
 #
-Tiger_FSScan_Setuid=Y		# Setuid executables
-Tiger_FSScan_Setgid=Y		# Setgid executables
+Tiger_FSScan_Setuid=N		# Setuid executables
+Tiger_FSScan_Setgid=N		# Setgid executables
 Tiger_FSScan_Devs=Y		# device files
 Tiger_FSScan_SymLinks=Y		# strange symbolic links
 Tiger_FSScan_ofNote=Y		# weird filenames
-Tiger_FSScan_WDIR=Y		# world writable directories
+Tiger_FSScan_WDIR=N		# world writable directories
 Tiger_FSScan_Unowned=Y		# files with undefined owners/groups
 #
 # Should we scan read-only filesystems
@@ -303,7 +319,7 @@
 # Note that .rhosts and .netrc need not appear here, as they will
 # be checked by scan_rhosts or scan_netrc.
 #
-USERDOTFILES=".alias .kshrc .cshrc .profile .login .mailrc .exrc .emacs .forward .tcshrc .zshenv .zshrc .zlogin .zprofile .rcrc .bashrc .bash_profile .inputrc .xinitrc .fvwm2rc .Xsession .Xclients .less"
+USERDOTFILES=".alias .kshrc .cshrc .profile .login .mailrc .exrc .emacs .forward .tcshrc .zshenv .zshrc .zlogin .zprofile .rcrc .bashrc .bash_profile .bash_logout .inputrc .xinitrc .fvwm2rc .Xsession .Xclients .less .kde"
 #
 # Rhost sites which are expected to be in the .rhosts files.
 # Anything that doesn't match will be reported.  The patterns
@@ -316,7 +332,8 @@
 # Debian GNU/Linux: default is 999, users are generated over 1000
 # Solaris: default should be 99, users are generated over 100
 # HP-UX (?): default should be 499, users are generated over 500
-Tiger_Accounts_Trust=999
+# Fedora : default should be 499, users are generated over 500
+Tiger_Accounts_Trust=499
 #
 # These SSH directive variables are used to specify "allowed" values
 # for the SSH Daemon.
@@ -324,7 +341,7 @@
 # be left blank to ignore the check.
 Tiger_SSH_Protocol='1|2'
 Tiger_SSH_RhostsAuthentication='no'
-Tiger_SSH_PasswordAuthentication='no'
+Tiger_SSH_PasswordAuthentication='yes|no'
 #
 # Should we give warnings on services that listen on all interfaces?
 # (i.e. those that have not been configured to listen only on one)
@@ -353,10 +370,16 @@
 # seen in the process table will generate a FAIL:
 #
 # The process list below is just an example (useful for Linux)
-# change it to suit your needs
-# Tiger_Running_Procs='syslogd cron atd klogd'
+# change it to suit your needs. You can use either the process name
+# or the full path name
+Tiger_Running_Procs='syslogd crond atd klogd'
+# or
+# Tiger_Running_Procs='/sbin/syslogd /usr/sbin/atd /usr/sbin/crond /sbin/klogd'
 #
 # Should we optimize DPKG checks? (by not using dpkg but looking on
 # the file database at /var/lib/dpkg?)
 #
 Tiger_DPKG_Optimize=Y
+
+# Devices are not only in /dev when udev is enabled :
+FS_DEVDIRS="/dev /etc/udev/devices"
--- ./tigerrc-all.config	2003-10-07 00:02:56.000000000 +0200
+++ ./tigerrc-all	2005-08-07 17:12:40.000000000 +0200
@@ -9,6 +9,7 @@
 #
 TigerNoBuild=Y			# C files are corrupted (ouch.)
 Tiger_Check_PASSWD=Y		# Fast
+Tiger_Check_PASSWD_FORMAT=N     # Fast - not needed if on systems with pwck
 Tiger_Check_PASSWD_SHADOW=Y	# Time varies on # of users
 Tiger_Check_PASSWD_NIS=N	# Time varies on # of users
 Tiger_Check_GROUP=Y		# Fast
@@ -274,6 +275,11 @@
 #
 Tiger_Crack_Local=Y
 #
+# Who sends output from 'tigercron'?
+# Default is "root@$HOSTNAME" (gets expanded by tigercron)
+#
+# Tiger_Mail_FROM="root@`uname -n`"
+#
 # Who gets output from 'tigercron'?
 #
 Tiger_Mail_RCPT=root
