--- tiger-3.2.1.orig/c/Makefile
+++ tiger-3.2.1/c/Makefile
@@ -1,3 +1,4 @@
+# Generated automatically from Makefile.in by configure.
 # Makefile for tiger binaries
 # 
 #
@@ -22,15 +23,17 @@
 # been added later)
 BINLIST=getpermit snefru md5 testsuid realpath
 COPTS= -DNEEDGETWD
+INSTALL=/usr/bin/install -c
+CC=gcc
 
 all: ${BINLIST}
 
 install:  all
-	cp ${BINLIST} ../bin/
+	$(INSTALL) -m0755 ${BINLIST} ../bin/
 
 
 % : %.c
-	gcc ${COPTS} -o $@ $<
+	$(CC) $(CFLAGS) ${COPTS} -o $@ $<
 
 clean:
 	-rm -f ${BINLIST}
--- tiger-3.2.1.orig/c/Makefile.in
+++ tiger-3.2.1/c/Makefile.in
@@ -0,0 +1,40 @@
+# Makefile for tiger binaries
+# 
+#
+#     Makefile for tiger binaries - A UN*X security checking system
+#     Copyright (C) 2002 Javier Fernandez-Sanguino Pen~a <jfs@computer.org>
+#
+#    This program is free software; you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation; either version 1, or (at your option)
+#    any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#     Please see the file `COPYING' for the complete copyright notice.
+#
+
+# For some reason snefru and md5 were received wrongly
+# and they could not be compiled at first (they have
+# been added later)
+BINLIST=getpermit snefru md5 testsuid realpath
+COPTS= -DNEEDGETWD
+INSTALL=@INSTALL@
+CC=@CC@
+
+all: ${BINLIST}
+
+install:  all
+	$(INSTALL) -m0755 ${BINLIST} ../bin/
+
+
+% : %.c
+	$(CC) $(CFLAGS) ${COPTS} -o $@ $<
+
+clean:
+	-rm -f ${BINLIST}
+
+distclean: clean
--- tiger-3.2.1.orig/configure.in
+++ tiger-3.2.1/configure.in
@@ -4,6 +4,9 @@
 dnl for to make sure it's in the right directory.
 AC_INIT(tiger)
 
+AC_PROG_CC
+AC_PROG_INSTALL
+
 test "x$prefix" != "xNONE" || prefix=/usr/local
 
 AC_ARG_WITH(tigerhome, [  --with-tigerhome=LOCATION        Location of all Tiger files],
@@ -42,6 +45,7 @@
 tigerlogdir=`eval echo $tigerlog`
 tigerconfigdir=`eval echo $tigerconfig`
 tigerhomedir=`eval echo $tigerhome`
+AC_SUBST(install)
 AC_SUBST(tigerhome)
 AC_SUBST(tigerwork)
 AC_SUBST(tigerlog)
@@ -52,5 +56,6 @@
 AC_SUBST(tigerconfigdir)
 AC_SUBST(tigerhomedir)
 AC_OUTPUT(Makefile
+	c/Makefile
 	man/tiger.8
 	man/tigercron.8)
--- tiger-3.2.1.orig/Makefile.in
+++ tiger-3.2.1/Makefile.in
@@ -21,6 +21,9 @@
 mandir=@mandir@
 # To avoid troubles with some systems..
 SHELL = /bin/sh
+# Installation program
+INSTALL=@INSTALL@
+CC=@CC@
 
 # This directory will contain the 'tiger', 'tigercron', 'tigexp'
 # scripts, config files, the 'scripts' subdirectory which will
@@ -54,6 +57,10 @@
 #
 TIGERCONFIG=@tigerconfig@
 #
+# Where do manpages go to
+# 
+TIGERMANDIR=@mandir@
+#
 #------------------------------------------------------------------------
 #
 # End of user customization...
@@ -61,7 +68,7 @@
 #------------------------------------------------------------------------
 #
 
-PLATFORM_SCRIPTS=$(shell find ./systems/ -type f)
+PLATFORM_SCRIPTS=$$(find ./systems/ -type f)
 
 BINARIES=./tiger \
 	 ./tigexp  \
@@ -89,6 +96,8 @@
 	./scripts/check_netrc \
 	./scripts/check_network \
 	./scripts/check_nisplus \
+	./scripts/check_ntp \
+	./scripts/check_omniback \
 	./scripts/check_passwd \
 	./scripts/check_passwdformat \
 	./scripts/check_path \
@@ -116,11 +125,13 @@
 	$(PLATFORM_SCRIPTS)
 
 CONFIGFILES=./tigerrc \
-	./cronrc \
+	./cronrc
 
 MISCFILES=./initdefs \
-	./check.tbl \
 	./syslist \
+	./check.d/README \
+	./util/buildbins \
+	./util/buildconf \
 	./util/difflogs \
 	./util/flogit \
 	./util/genmsgidx \
@@ -130,19 +141,29 @@
 	./util/getfs-std \
 	./util/gethostinfo \
 	./util/getnetgroup \
+	./util/getpermit \
+	./util/installsig \
 	./util/logit \
+	./util/mkfilelst \
+	./util/mksig \
 	./util/setsh \
 	./util/sgrep
 
 MISCDIRS=./bin \
+	./check.d \
 	./doc \
 	./html \
-	./man \
 	./scripts/sub \
 	./systems
 
 all:
+	@if [ ! -d bin ]; then \
+	  mkdir bin; \
+	fi
 	cd c && $(MAKE) install
+	@if [ ! -d html ]; then \
+	  mkdir html; \
+	fi
 	cd util && sh doc2html
 	./util/genmsgidx
 
@@ -182,6 +203,19 @@
 	  mkdir -p $(DESTDIR)$(TIGERCONFIG); \
 	  chmod 700 $(DESTDIR)$(TIGERCONFIG); \
 	fi
+	@echo "Creating $(TIGERMANDIR)..."
+	@if [ ! -d $(DESTDIR)$(TIGERMANDIR) ]; then \
+	  mkdir -p $(DESTDIR)$(TIGERMANDIR); \
+	  mkdir -p $(DESTDIR)$(TIGERMANDIR)/man8; \
+	  chmod 755 $(DESTDIR)$(TIGERMANDIR); \
+	  chmod 755 $(DESTDIR)$(TIGERMANDIR)/man8; \
+	fi
+
+installmanpages:
+	@echo "Copying manpages..."
+	$(INSTALL) -m 0444 man/tiger.8 $(DESTDIR)$(TIGERMANDIR)/man8
+	$(INSTALL) -m 0444 man/tigexp.8 $(DESTDIR)$(TIGERMANDIR)/man8
+	$(INSTALL) -m 0444 man/tigercron.8 $(DESTDIR)$(TIGERMANDIR)/man8
 
 installbinaries:
 	@echo "Copying binaries..."
@@ -198,8 +232,7 @@
 installconfig:
 	@echo "Copying configuration files..."
 	@for file in $(CONFIGFILES); do \
-	  cp $$file $(DESTDIR)$(TIGERCONFIG)/$$file; \
-	  chmod 640 $(DESTDIR)$(TIGERCONFIG)/$$file; \
+	  $(INSTALL) -m 0640 $$file $(DESTDIR)$(TIGERCONFIG)/$$file; \
 	done
 	@echo "Copying general configuration..."
 	@sed -e 's%^TigerLogDir=.*$$%TigerLogDir="'$(TIGERLOGS)'"%' \
@@ -210,13 +243,13 @@
 	 ./config >$(DESTDIR)$(TIGERHOME)/config
 	@chmod 644 $(DESTDIR)$(TIGERHOME)/config
 
-install: installdirs installbinaries installconfig
+install: installdirs installbinaries installconfig installmanpages
 	cd c && $(MAKE) install
-	@echo "Copying miscellaneus dirs..."
+	@echo "Copying miscellaneous dirs..."
 	@for dir in $(MISCDIRS); do \
 	  tar cf - $$dir | (cd $(DESTDIR)$(TIGERHOME); tar xpf -); \
 	done
-	@echo "Copying miscellaneus files..."
+	@echo "Copying miscellaneous files..."
 	@for file in $(MISCFILES); do \
 	  cp -p $$file $(DESTDIR)$(TIGERHOME)/$$file; \
 	done
